### PR TITLE
add on the fly video recording and uploading

### PR DESCRIPTION
**Pull request by flaviogrossi**
@RocketChat/core 

The following pr adds the possibility to record a video from the webcam and upload it, just like it is currently possible for audio messages.

This patch adds a new button in the message box (visible only if the relevant setting is checked in the preferences) which opens a popup window with the webcam preview and allows to record and upload the video.

camera_dialog

After this, the usual file upload dialog is showed (the file upload settings must be set to accept the video/webm mime type) and the possibility to preview the recorded video has been added.

upload_dialog

Please let me know what do you think and if any change is needed in the code.

### CODE REVIEW

File: packages/rocketchat-ui-vrecord/package.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+	api.use([
+		'mongo',
+		'ecmascript',
+		'templating',
+		'coffeescript',
+		'underscore',
+		'tracker',
+		'rocketchat:lib',
+		'less@2.5.1'
```
Comment (rodrigok) [inline packages/rocketchat-ui-vrecord/package.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2754409530/issues/source?bid=3199634&fileBranchId=3358103#l19)

File: packages/rocketchat-ui-vrecord/package.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Package.onUse(function(api) {
+	api.versionsFrom('1.2.1');
+
+	api.use([
+		'mongo',
+		'ecmascript',
+		'templating',
+		'coffeescript',
+		'underscore',
+		'tracker',
```
Comment (rodrigok) [inline packages/rocketchat-ui-vrecord/package.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2754409530/issues/source?bid=3199634&fileBranchId=3358103#l17)

File: packages/rocketchat-ui-vrecord/package.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+		'rocketchat:lib',
+		'less@2.5.1'
+	]);
+
+
+	api.addAssets('client/vrecord.less', 'server');
+
+	api.addFiles('client/vrecord.html', 'client');
```
Comment (rodrigok) [inline packages/rocketchat-ui-vrecord/package.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2754409530/issues/source?bid=3199634&fileBranchId=3358103#l25)

File: packages/rocketchat-ui-vrecord/package.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	api.addFiles('client/vrecord.coffee', 'client');
+	api.addFiles('client/VRecDialog.coffee', 'client');
```
Comment (rodrigok) [inline packages/rocketchat-ui-vrecord/package.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2754409530/issues/source?bid=3199634&fileBranchId=3358103#l27)

File: packages/rocketchat-ui-vrecord/package.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	"version": "0.0.1",
+	"description": "Video upload with on the fly recording",
+	"documentation": "README.md"
+});
+
+Package.onUse(function(api) {
+	api.versionsFrom('1.2.1');
+
+	api.use([
+		'mongo',
```
Comment (rodrigok) [inline packages/rocketchat-ui-vrecord/package.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2754409530/issues/source?bid=3199634&fileBranchId=3358103#l12)

File: packages/rocketchat-ui-vrecord/package.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+		'tracker',
+		'rocketchat:lib',
+		'less@2.5.1'
+	]);
+
+
+	api.addAssets('client/vrecord.less', 'server');
+
+	api.addFiles('client/vrecord.html', 'client');
+	api.addFiles('client/vrecord.coffee', 'client');
```
Comment (rodrigok) [inline packages/rocketchat-ui-vrecord/package.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2754409530/issues/source?bid=3199634&fileBranchId=3358103#l26)

File: packages/rocketchat-ui-vrecord/package.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+Package.describe({
+	"name": "rocketchat:ui-vrecord",
+	"version": "0.0.1",
+	"description": "Video upload with on the fly recording",
+	"documentation": "README.md"
+});
+
+Package.onUse(function(api) {
+	api.versionsFrom('1.2.1');
```
Comment (rodrigok) [inline packages/rocketchat-ui-vrecord/package.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2754409530/issues/source?bid=3199634&fileBranchId=3358103#l9)

File: packages/rocketchat-ui-vrecord/package.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+	api.use([
+		'mongo',
+		'ecmascript',
+		'templating',
```
Comment (rodrigok) [inline packages/rocketchat-ui-vrecord/package.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2754409530/issues/source?bid=3199634&fileBranchId=3358103#l14)

File: packages/rocketchat-ui-vrecord/package.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+		'coffeescript',
+		'underscore',
```
Comment (rodrigok) [inline packages/rocketchat-ui-vrecord/package.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2754409530/issues/source?bid=3199634&fileBranchId=3358103#l16)

File: packages/rocketchat-ui-vrecord/package.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+	]);
+
+
+	api.addAssets('client/vrecord.less', 'server');
+
+	api.addFiles('client/vrecord.html', 'client');
+	api.addFiles('client/vrecord.coffee', 'client');
+	api.addFiles('client/VRecDialog.coffee', 'client');
+
+	api.addFiles('server/settings.coffee', 'server');
```
Comment (rodrigok) [inline packages/rocketchat-ui-vrecord/package.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2754409530/issues/source?bid=3199634&fileBranchId=3358103#l29)

**Comment by geekgonecrazy** [comment]
@flaviogrossi cool!  Thanks for Contributing.  The team is reviewing 👍

**Comment by graywolf336** [comment]
@flaviogrossi does this work on mobile as well?

**Comment by flaviogrossi** [comment]
> does this work on mobile as well?

opening the webcam works fine on android by adding this plugin:
cordova-plugin-chrome-apps-videocapture

the MediaRecorder symbol though is undefined, i don't think it is supported by cordova.

**Comment by rodrigok** [comment]
LGTM

**Comment by geekgonecrazy** [comment]
LGTM

**Comment by graywolf336** [comment]
@flaviogrossi Mind fixing the conflicts so this can be merged? :)

**Comment by flaviogrossi** [comment]
> Mind fixing the conflicts so this can be merged? :)

@graywolf336 done

**Pull request merged**